src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/ListKeyValueSelectorExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Models;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    internal static class ListKeyValueSelectorExtensions
```

Comment:
I think we may be able to generalize the concept of preserving precedence order while maintaining unique elements in the IEnumerable collection. What do you think about making this `IEnumerableExtensions` with the following extension method:

`public static void AppendUnique<T>(this IEnumerable<T> selectors, T selector)`

Reply:
> public static void AppendUnique<T>(this IEnumerable<T> selectors, T selector)

I was considering that before. then we might also need to specify a `Equals` method for T? BTW Cons for above util:
- Calling this function, we would first construct a new instance of the selector, regardless of whether it's duplicate. Not sure if the slight increase of memory consumption is important or not.

Reply:
We dont necessarily need equals method because we can still check all 3 properties manually. But I'm also not against adding Equals override for `KeyValueSelector` since we have precedence for doing the same in `KeyValueWatcher` and `KeyValueIdentifier`.

About the memory consumption, I think KeyValueSelector is a lightweight object with just 2 (or 3) string values. Since we are not persisting these instances of `KeyValueSelector`, memory consumption doesn't sound like a big concern to me.

Reply:
Then we might use `List` instead of `IEnumerable` as it doesn't have `Add` or `Remove`...

Reply:
updated using `public static void AppendUnique<T>(this List<T> items, T item)`

Note, since we have the generic util, I'm also using it to dedup multiKeyWatchers.